interview-questions.md

```diff
@@ -1,57 +1,83 @@
+
+Researched answer: The .map built in method takes in 3 parameters are arguements, in one individual data set. It takes the same 3 arguements are .forEach but its reutrn a new array. [value, index, array] Index and Array are optional components in the data set.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: .map built in method with return the array to the original length of the inputted array. The .filter function is looking to minimize/simply the array by using what logic is inputted into the parameter. The .map will take logic the same as .filter but will not show the same results as .filter.
+
+Researched answer: The .map and .filter are both built in methods that perform specificed logic to create a desired outcome/output. The difference between the two is that .map will return the original size of the parameter, and .filter will return the a new array with only displaying the results. .filter is looking to shorten the array.
 
-Researched answer:
 
 4. What is iteration?
 
-Your answer:
+Your answer: A set of instructions that are expected to repeat until stopped.
+
+Researched answer:  Itireation are specific set of instructions that are executed over and over until a condition is met. Arrays, strings and objects can perform iteration. An iteration needs a start, where to stop counting, and how to get from the start to the stop.
+
 
-Researched answer:
 
 5. What is the difference between a class and an object?
 
-Your answer:
+Your answer: A class are the blueprints of an object blue prints often have small spacific details and instructions on how things are suppose to be done as a whole they make up an object. A object is a data type and with in it, it has behavior. 
 
-Researched answer:
+Researched answer: Classes are the blueprint for an object, it provides plans and specifications for an object. Classes are the brains whereas objects is the braun. A class can give attributes and properties. such as Name, Age, Gender, As for an object it would be the thing i am using my specific class for. FUN FACT: When a function belongs to an object it is called a method. Within these methods we can get more detailed on what we can call baised on the perameters of an object   
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: Hoisting in javascript is when you take a collection of parameters in one javascript file, and move them to another.
 
-Researched answer:
+Researched answer: Javascript hoisting is the process in which you can use javascript's function and behaviors prior to them being declared. It allows us to take variables and functions, no matter where they are declared, and they are moved to the top.
```

Comment:
Overall, great job in this section.  There are a few instances where your initial answer was a little short, so keep working on expanding those answers to really show your interviewer what you know.